solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1838,6 +1838,22 @@ private void checkStateInZk(CoreDescriptor cd) throws InterruptedException, NotI
     }
   }
 
+  public void tryCancelAllElections() {
+    if (zkClient.isClosed()) {
+      return;
+    }
+    electionContexts.values().parallelStream().forEach(context -> {
```

Comment:
How many contexts do we expect here. Does it actually make sense to do a parallel stream?

Reply:
No, it does not. Parallelstream was being used for close in ZkController.close() as well so this was basically copied code. However I checked and it is a lightweight operation. It basically removes a zk node and sets a volatile member. I have replaced this call with a serial forEach in both places.

@hossman pointed out that this iteration wasn't safe so I have put the forEach inside a synchronized block